### DISCUSSION TITLE
create() in tutorial not working

### DISCUSSION THREAD

**Discussion by mattewkl** [pallets/flask#4471] [Q&A]
So, i am trying to write simple tutorial app with flask, and on create() function in blog.py its completly stuck. Whenever i try to create new post, **TypeError: create() takes 0 positional arguments but 2 were given
The view function did not return a valid response. The return type must be a string, dict, tuple, Response instance, or WSGI callable, but it was a function** occures. And i dont have any clue what i did wrong, because i was following instructions, and after several tries i gave up and COPY  all the code. And it still not working. I need help ASAP pls. current progress in tutorial - https://github.com/mattewkl/flask_tut
Thanks in advance for any help

**Comment by cibinmathew** [comment]
How did you run
Can you post detailed error/snapshots

**Reply by mattewkl** [reply]
ran just like tutorial suggested (command line)
<img width="960" alt="create_bug" src="https://user-images.githubusercontent.com/93080414/156142551-4119010b-d828-4a9c-9f84-c8c17727f036.png">

**Reply by cibinmathew** [reply]
pls post the tutorial link referred

**Reply by mattewkl** [reply]
https://flask.palletsprojects.com/en/2.0.x/tutorial/blog/ is the step where error occured

**Comment by cibinmathew** [comment]
Pls check this once if not tried already
https://github.com/pallets/flask/tree/2.0.3/examples/tutorial

**Reply by mattewkl** [reply]
I have already copyed code from there

**Reply by cibinmathew** [reply]
I am able to reproduce the error now.

**Reply by mattewkl** [reply]
Any clue what it reffered to?

**Comment by cibinmathew** [comment]
Some difference in the code against the original https://github.com/mattewkl/flask_tut/blob/main/flaskr/auth.py#L102

**Reply by cibinmathew** [reply]
Original source for reference: https://github.com/pallets/flask/blob/2.0.3/examples/tutorial/flaskr/auth.py#L27

**Reply by mattewkl** [reply]
Is this related to error?

**Reply by mattewkl** [reply]
IT IS! My god, thank you!

**Reply by cibinmathew** [reply]
glad it helped. you can close the thread
